core/src/main/scala/epollcat/internal/ch/SocketHelpers.scala

```diff
@@ -89,4 +89,29 @@ private[ch] object SocketHelpers {
     new InetSocketAddress(inetAddr, port)
   }
 
+  /* Return text translation of getaddrinfo (gai) error code.
+   * Since this is predominantly executed whilst on an error path,
+   * be particularly picky about checking for nulls. Prior code on the path
+   * should have reported or thrown NPE on nulls. But here we are, probably
+   * on an error path with an indeterminate world.
+   *
+   * The extra 'address' and 'port' information is given to possibly
+   * provide clues to reduce defect removal cycles.
+   */
+  def getGaiErrorMessage(gaiErrorCode: CInt, addr: InetSocketAddress): String = {
+    java.util.Objects.requireNonNull(addr, "unexpected null InetSocketAddress")
+
+    val gaiMsg = fromCString(posix.netdb.gai_strerror(gaiErrorCode))
+
+    val adr = {
+      val aga = addr.getAddress()
+      if (aga == null) "null" // No requireNonNull, salvage port info
+      else aga.getHostAddress()
+    }
+
+    val port = addr.getPort.toString
+
+    s"${gaiMsg} address: ${adr}  port: ${port}"
```

Comment:
What are your thoughts on the security concersn with including the `address` and `port` in the error message? We recently had this exact discussion on the Typelevel Discord.

Some insights from Michael Pilquist:
> any security concerns about including host/port in the error message? e.g., imagine an http4s route that tries to make a request to some other service and fails with timeout -- that exception message would get propagated to client, exposing internal hosts/port info

Reply:
Thank you for the review & raising the concern.

TL;DR
If there is any chance that the info will impede upstream adoption, I will remove
it and use a private version.  No problem.

Longer discussion:

Personal opinion:

Bad actors are incredibly clever at exploiting _any_ lack of entropy. Witness all the
Intel & AMD speculative execution hacks.

People trying to fix problems depend upon there being a lack of entropy: i.e. information.

The very fact that a getaddrinfo message of any kind is being reported gives
suitably motivated bad actors information.  Removing even a simple message
makes life difficult/impossible for the "fixers".

"fixers" can be overwhelmed by too much information, so finding the appropriate
amount is a changing goal.

Here the port number probably gives away the most information.
The address is in n.n.n.n or (soon) xxxx::x notation, not a string
"secrets.com". The numeric address can be translated, but
it takes someone's time and a bit of skill.  

The understanding of "Industry best-practice" changes over time.
At this point and for the foreseeable future I think giving at
least the basic getaddrinfo code translation can be defended
as "Industry standard".

I think we concur that `Rapid prototyping` is not an occasion for a security give-away.
Security analysis is essential but needs to be considered in scope: valid security, extra-cautious security,
paranoid security, and security-theater. 

I think the base PR is safe but not worth investing the time to defend to the
"deny any information at all" folks.

Reply:
Thanks for sharing your thoughts. Unfortunately I am not a security expert, so in doubt I defer to the JVM: can we simulate some of these errors in our test suite, and make the exceptions match what the JVM does in the same situation.

Reply:
Looks like Java 17 ServerSocket `bind` distinguishes EAI_FAMILY (unsupported address family)
and throws an `IllegalArgumentException`. In other cases it throws `IOException`. The
current epoll code always throws `IOException` (with a message that would give the unsupported
AF, but probably not tell which af it got).  That is fixable.

I can try to provoke the a SocketServer bind IOException and see exactly what
text it returns.  Will take time (day or two-ish).

Then I have to check the java.nio Channel area to see if there is any 
complication/change in message.   

I can submit the simple (no address, no port) gai_message updates
so they do not get lost on my end. Understanding that they are on 
hold whilst I find out what Java does.

Reply:
Thanks for understanding! That sounds like a good plan to me.

Reply:
Update:

Java 17 AsynchronousSeverSocketChannel throws `UnsupportedAddressTypeException` not `IllegalArgumentException`
for wrong AF.  I expect AsynchronousSocketChannel to to the same but will check before implementing.